### PR TITLE
Require Java 8, and always require the annotated JDK

### PR DESCRIPTION
**Pull request by mernst**

### CODE REVIEW

File: framework/src/org/checkerframework/framework/source/SourceChecker.java [hunk 1]
```diff
@@ -427,6 +428,15 @@ public final synchronized void init(ProcessingEnvironment env) {
...
         // This is used to trigger AggregateChecker's setProcessingEnvironment.
         setProcessingEnvironment(env);
+
+        double jreVersion = PluginUtil.getJreVersion();
+        if (jreVersion != 1.8) {
+            report(
+                    org.checkerframework.framework.source.Result.failure(
+                            "The Checker Framework must be run under JDK 1.8.  You are using version %f.",
+                            jreVersion),
+                    null);
```
Comment (mernst) [inline framework/src/org/checkerframework/framework/source/SourceChecker.java]: This uses `null` as the location at which to issue the warning.
Is there a better way to issue the warning, such as by immediately terminating compilation?

Comment (smillst) [reply framework/src/org/checkerframework/framework/source/SourceChecker.java]: Yes, use ErrorReporter.errorAbort().  That method throws an exception.

**Comment by mernst** [comment]
@smillst This pull request is still failing because many tests seem not to use the annotated JDK.
I think probably those tests should use the annotated JDK, but an alternative would be to give a way to override the Checker Framework's requirement to use the annotated JDK.

**Comment by smillst** [comment]
Framework tests need 'nocheckjdk' arguments.

**Comment by mernst** [comment]
@smillst Framework tests need to pass some "don't check for JDK" flag.

File: checker/src/org/checkerframework/checker/nullness/KeyForAnnotatedTypeFactory.java [hunk 2]
```diff
@@ -93,7 +93,7 @@ protected TypeHierarchy createTypeHierarchy() {
         return new KeyForTypeHierarchy(
                 checker,
                 getQualifierHierarchy(),
-                checker.getOption("ignoreRawTypeArguments", "true").equals("true"),
+                checker.getBooleanOption("ignoreRawTypeArguments"),
```
Comment (smillst) [inline checker/src/org/checkerframework/checker/nullness/KeyForAnnotatedTypeFactory.java]: I think this pull request contains changes that are also in #1891.

**Comment by mernst** [comment]
@smillst I think I addressed your comments. Can you see whether this is ready to be merged?  Thanks!

**Pull request merged**